File: lista-de-exercicios/exercicio-4/Calculadora.ts
import { Menu } from './Menu.js';

export class Calculadora {
    private operacao: string;
    private numero1: number;
    private numero2: number;

    public calcular(
        numero1: number,
        numero2: number,
        operacao: string): number {
        if (!numero1 || !numero2)
            throw Error ('Número inválido');
        switch (operacao) {
            case '+':
                return numero1 + numero2;
            case '-':
                return numero1 - numero2;
            case '*':
                return numero1 * numero2;
            case '/':
                return numero1 / numero2;
            case '^':
                return Math.pow(numero1, numero2);
            default: 
                throw Error ('Operador invalido');
        }
    }
}

File: lista-de-exercicios-java/src/main/java/org/example/exercicio04/Calculadora.java
package org.example.exercicio04;

public class Calculadora {
    private char operacao;
    private double numero1;
    private double numero2;

    public double calcular(double numero1, char operacao, double numero2) {
        if (operacao == '/' && numero2 == 0)
            throw new IllegalArgumentException("Número inválido");

        this.numero1 = numero1;
        this.numero2 = numero2;
        this.operacao = operacao;

        switch (operacao) {
            case '+':
                return numero1 + numero2;
            case '-':
                return numero1 - numero2;
            case '*':
                return numero1 * numero2;
            case '/':
                return numero1 / numero2;
            case '^':
                return Math.pow(numero1, numero2);
            default:
                throw new IllegalArgumentException("Operador inválido");
        }
    }
}


